test/src/lib.rs

```diff
@@ -86,44 +101,70 @@ impl CweTestCase {
 
 /// Mark test cases using the given CPU architecture as `skipped`.
 pub fn mark_architecture_skipped(test_cases: &mut [CweTestCase], arch: &str) {
-    for test in test_cases.iter_mut() {
-        if test.architecture == arch {
-            test.skipped = true;
-        }
-    }
+    mark_skipped_closure(test_cases, |test| test.architecture == arch)
 }
 
 /// Mark test cases using the given compiler as `skipped`.
 pub fn mark_compiler_skipped(test_cases: &mut [CweTestCase], comp: &str) {
-    for test in test_cases.iter_mut() {
-        if test.compiler == comp {
-            test.skipped = true;
-        }
-    }
+    mark_skipped_closure(test_cases, |test| test.compiler == comp)
 }
 
 /// Mark test cases using the given CPU architecture + compiler combination as `skipped`.
 pub fn mark_skipped(test_cases: &mut [CweTestCase], value1: &str, value2: &str) {
-    for test in test_cases.iter_mut() {
-        if (test.architecture == value1 && test.compiler == value2)
+    mark_skipped_closure(test_cases, |test| {
+        (test.architecture == value1 && test.compiler == value2)
             || (test.architecture == value2 && test.compiler == value1)
-        {
+    })
+}
+
+/// Mark test cases using the given CPU architecture + compiler combination as `skipped`
+/// iff they are not Linux kernel modules.
+pub fn mark_skipped_user(test_cases: &mut [CweTestCase], value1: &str, value2: &str) {
+    mark_skipped_closure(test_cases, |test| {
+        !test.is_lkm
+            && ((test.architecture == value1 && test.compiler == value2)
+                || (test.architecture == value2 && test.compiler == value1))
+    })
+}
+
+fn mark_skipped_closure<F>(test_cases: &mut [CweTestCase], predecate: F)
```

Comment:
typo: `predecate` -> `predicate`
Also, I prefer giving all functions at least a very short doc comment, even small helper functions like this.